.github/workflows/test-convene-web.yml

```diff
@@ -2,8 +2,68 @@ name: Test
 on: push
 
 jobs:
+  build:
+    name: Build app
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      # Temporary mitigation for https://github.com/actions/runner-images/issues/6283
+      # Perhaps we should consider not using brew in CI (though it sure is nice!)
+
+      - name: Set up Homebrew environment
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Setup Ruby and install gems
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - name: Setup CI database.yml
+        run: cp config/database.yml.github-actions config/database.yml
+
+      - name: Use Development mode env
+        run: cp .env.example .env
+
+      # Yarn caching steps borrowed from https://github.com/actions/cache/blob/main/examples.md#node---yarn
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Get brew cache path
+        id: brew-cache-path
+        run: echo "::set-output name=path::$(brew --cache)"
+
+      - name: Configure Homebrew cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.brew-cache-path.outputs.path }}
+          key: ${{ runner.os }}-brew-${{ hashFiles('bin/setup') }}
+          restore-keys: ${{ runner.os }}-brew-
+
+      - name: Setup test environment
+        run: |
+          sudo chown -R $USER /usr/local/lib/node_modules
+          sudo chown -R $USER /usr/local/bin/
+          bin/setup
+
+      - name: Start Rails server
+        run: |
+          export PATH=$PATH:~/bin
+          bin/run &
```

Comment:
note: `build` job currently has the same steps that `test` job does. but this is s WIP and I'll see if I can make setup a reusable step to DRY it up. Since we aren't using docker-compose to build the full stack i dunno it it's possible to build the app once and then share it between the different jobs. each job runs on its own VM so I think it's like each job is running on a fresh machine.

Reply:
It looks like you're [already setting up caching](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows), which should make downstream containers "speedrun" the bundle/yarn/brew processes!

So the most likely vector of decomposition would be to gather the duplicative bits [into a job](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow) or [pulling out a workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) but I'm not really fluent enough in Github Action's domain specific language to know which pathway is most likely to be shortest-to-successful.